### ISSUE TITLE
Don't report 99 MBit on a 100-200Mbit connection as 'BAD'.

### ISSUE THREAD

**Issue by nihilus** [dotse/bbk#3]
There need to be some sort of limit based in percentage of the connection to determine the speed, not just some hard limits.

**Comment by arfcarl** [comment]
Have you checked the Readme? https://frontend.bredbandskollen.se/download/README.txt

'If the subscription can be identified, the download result will be evaluated as
"GOOD", "ACCEPTABLE", or "BAD". The result is considered BAD if the speed is
below the lower end of the expected speed range as stated by the service
provider. It is considered ACCEPTABLE if it is above the lower end, but still
clearly below the middle, of the expected speed range. If the result is better,
it is considered GOOD.
E.g. if the expected range is 150-250 Mbit/s, a result below 150 is BAD; a
result between 150 and 190 is ACCEPTABLE, and a result above 190 is GOOD.'

The "limit based in percentage" you're looking for is the 40% target that decides what is needed to get a GOOD result: 250 - 150 = 100 Mbit. 40% of 100 Mbit is 40 Mbit. 150 + 40 = 190 Mbit.

**Comment by nihilus** [comment]
Exactly and that's wrong in so many ways. 99,9% is NEVER a "bad" speed which implies it's unacceptable and hence makes people call customer support.

**Comment by jansaell** [comment]
We are getting the connection from the ISP - if we have the possibility.
If you ISP says you connection should be 100-200 MB and you don't get up to the lowest limit we do consider that bad.

**Comment by nihilus** [comment]
> We are getting the connection from the ISP - if we have the possibility.
> If you ISP says you connection should be 100-200 MB and you don't get up to the lowest limit we do consider that bad.

And that is flawed since it's usually "up to" in the agreements and there might be other circumstances that lowers the speed. To actually know the connection quality you would have to have iperf server instances at the ISPs.

This creates confusion which creates unnecessary support calls for those working in the field.

**Comment by pv2b** [comment]
I just randomly found this issue, and I'd just like to say this makes no sense.

If the ISP considers 99 Mbit/s an acceptable speed on the connection that they sell, they should then sell the connection as a 99-200 Mbit/s connection, not a 100-200 Mbit/s connection.

If the ISP promises 100-200 Mbit/s, then anything below 100 Mbit/s is unacceptable. (Assuming, for now, that the measurement server is well enough connected to the ISP's network to be a valid measurement of the connection speed itself.)

Imagine, if I went to the supermarket, and I bought a box of milk that is specified to be 100-200 cl, and then I come home and see I only got 99 cl. In this case, the amount of milk that I received is out of spec, even if it's only 1% out of spec.
